v1/remote/image_test.go

```diff
@@ -0,0 +1,82 @@
+package remote
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/authn"
+	"github.com/google/go-containerregistry/name"
+)
+
+type mockRegistry struct {
+	requests []*http.Request
+}
+
+// TODO(jonjohnsonjr): Replace this with better test data.
+const (
+	goodManifest = `{
+  "schemaVersion": 2,
+  "config": {
+    "digest": "sha256:deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef"
+  }
+}`
+
+	goodConfig = `{
+  "architecture": "amd64"
```

Comment:
Check that `ConfigName` == `sha256(this)`?

Reply:
Done